go/interpreter/lfvm/opcode_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package lfvm
+
+import (
+	"math"
+	"regexp"
+	"slices"
+	"testing"
+)
+
+func TestOpcode_UnnamedOpcodes(t *testing.T) {
+	validName := regexp.MustCompile(`^0x00[0-9A-Fa-f]{2}$`)
+	for i := 0; i < math.MaxUint16; i++ {
+		op := OpCode(i)
+		if !validName.MatchString(op.String()) && op > NUM_OPCODES {
+			t.Errorf("Invalid print for op %v (%d)", op, i)
+		}
+	}
+}
+
+func TestOpcode_NamedOpcodes(t *testing.T) {
+	validName := regexp.MustCompile(`^0x00[0-9A-Fa-f]{2}$`)
+	for i := 0; i < math.MaxUint16; i++ {
+		op := OpCode(i)
+		// NUM_EXECUTABLE_OPCODES is a special case, it does not have a string name
+		if validName.MatchString(op.String()) && op < NUM_OPCODES && op != NUM_EXECUTABLE_OPCODES {
```

Comment:
As mentioned before, the contiguous property of opcodes may change.
Not sure if this is easy to model somehow else at this point.

Reply:
as an answer to both your comments, I cannot make a test for a property that might change in the future. I can only make a test for existing code. Currently the constant `NUM_OPCODES` and `NUM_EXECUTABLE_OPCODES` are the way to distinguish whether an opcodes is valid or executable or neither, and this dictates what style of printing it should use. Given that there is also no bidirectional mapping to `tosca.OpCodes`. I'm not sure this can be tested otherwise